docs/tutorial/write-a-machine-charm-for-tinyproxy.md

```diff
@@ -0,0 +1,980 @@
+    """Return whether the tinyproxy executable is available."""
+    return shutil.which("tinyproxy") is not None
+
+
+def is_running() -> bool:
+    """Return whether tinyproxy is running."""
+    return bool(_get_pid())
+
+
+def reload_config() -> None:
+    """Ask tinyproxy to reload config."""
+    pid = _get_pid()
+    if not pid:
+        raise RuntimeError("tinyproxy is not running")
+    # Sending signal SIGUSR1 doesn't terminate the process. It asks the process to reload config.
+    # See https://manpages.ubuntu.com/manpages/jammy/en/man8/tinyproxy.8.html#signals
+    os.kill(pid, signal.SIGUSR1)
+
+
+def start() -> None:
+    """configure_and_runproxy."""
+    subprocess.run(["tinyproxy"], check=True, capture_output=True, text=True)
+
+
+def stop() -> None:
+    """Stop tinyproxy."""
+    pid = _get_pid()
+    if pid:
+        os.kill(pid, signal.SIGTERM)
+
+
+def uninstall() -> None:
+    """Uninstall the tinyproxy executable and remove files."""
+    apt.remove_package("tinyproxy-bin")
+    pathops.LocalPath(PID_FILE).unlink(missing_ok=True)
+    pathops.LocalPath(CONFIG_FILE).unlink(missing_ok=True)
+    pathops.LocalPath(CONFIG_FILE).parent.rmdir()
+
+
+def _get_pid() -> int | None:
+    """Return the PID of the tinyproxy process, or None if the process can't be found."""
+    if not pathops.LocalPath(PID_FILE).exists():
+        return None
+    pid = int(pathops.LocalPath(PID_FILE).read_text())
+    try:
+        # Sending signal 0 doesn't terminate the process. It just checks whether the PID exists.
+        os.kill(pid, 0)
+    except ProcessLookupError:
+        return None
+    return pid
+```
+
+Notice that the helper module is stateless. In fact, your charm as a whole will be stateless. The main logic of your charm will:
+
+1. Receive an event from Juju.
+2. Use the functions in the helper module to manage tinyproxy and check its status.
+3. Report the status back to Juju.
+
+```{tip}
+After adding code to your charm, run `tox -e format` to format the code. Then run `tox -e lint` to check the code against coding style standards and run static checks.
+```
+
+### Handle the install event
+
+We'll now write the charm code that handles events from Juju. Charmcraft created `src/charm.py` as the location for this logic, containing a class called `TinyproxyCharm`. We'll refer to `TinyproxyCharm` as the "charm class".
+
+In `src/charm.py`, replace the `_on_install` method of the charm class with:
+
+```python
+    def _on_install(self, event: ops.InstallEvent) -> None:
+        """Install tinyproxy on the machine."""
+        if not tinyproxy.is_installed():
+            tinyproxy.install()
+            version = tinyproxy.get_version()
+            self.unit.set_workload_version(version)
+```
+
+When your charm receives the "install" event from Juju, Ops runs this method and responds to Juju with the version of tinyproxy that's installed on the machine. Juju shows the tinyproxy version in its status output.
```

Comment:
"responds to Juju" doesn't seem right to me - it's not the "response" of the install event. What about something like "and tells Juju the version", or "responds by telling Juju the version".